src/quadiron-c.cpp

```diff
@@ -0,0 +1,450 @@
+/* -*- mode: c++ -*- */
+/*
+ * Copyright 2017-2018 Scality
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors
+ *    may be used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+#include "property.h"
+#include "quadiron-c.h"
+#include "quadiron.h"
+
+extern "C" {
+
+/**
+ * Create FNT FEC - This FEC is relatively complex because it requires storing a
+ * metadata header.
+ *
+ * @param word_size FNT only supports 1 or 2
+ * @param n_data number of data fragments
+ * @param n_parities number of parity fragments
+ *
+ * @return the FEC instance pointer
+ */
+struct QuadironFnt32*
+quadiron_fnt32_new(int word_size, int n_data, int n_parities, int systematic)
+{
+    size_t pkt_size = 1024;
+
+    if (word_size <= 4) {
+        return reinterpret_cast<struct QuadironFnt32*>(
+            new quadiron::fec::RsFnt<uint32_t>(
+                systematic ? quadiron::fec::FecType::SYSTEMATIC
```

Comment:
C also has a standard bool type since 1999, so it's safe to assume that after ~20 years we can start using bool in C.

Does liberasurecode only compile to C89?